apps/examples/binary_update/binary_update_main.c

```diff
@@ -78,20 +110,138 @@ static void binary_update_getinfo_all(void)
 	}
 }
 
-static void binary_update_getinfo(char *name)
+static void binary_update_check_valid_update(binary_info_t *pre_bin_info, binary_info_t *cur_bin_info, int condition)
+{
+	printf(" === [%5s] Update info === \n", cur_bin_info->name);
+	printf(" %4s | %8s | %s \n", "Con", "Version", "Inactive part");
+	printf(" ------------------------- \n");
+	printf(" %4s | %8s | %s \n", "Pre", pre_bin_info->version, pre_bin_info->dev_path);
+	printf(" %4s | %8s | %s \n", "Cur", cur_bin_info->version, cur_bin_info->dev_path);
+	printf(" ========================= \n");
+
+	if (condition == DOWNLOAD_VALID_BIN) {
+		if (strncmp(pre_bin_info->dev_path, cur_bin_info->dev_path, sizeof(pre_bin_info->dev_path) == 0)) {
+			fail_cnt++;
+			printf("Fail to load valid higher version binary.\n");
+		} else {
+			printf("Success to load valid higher version binary.\n");
+		}
+	} else { //DOWNLOAD_INVALID_BIN
+		if (strncmp(pre_bin_info->dev_path, cur_bin_info->dev_path, sizeof(pre_bin_info->dev_path) != 0)) {
+			fail_cnt++;
+			printf("Warning! Load invalid binary.\n");
+		} else {
+			printf("No update with invalid binary.\n");
+		}
+	}
+}
+
+static void binary_update_download(char *devname, int condition)
```

Comment:
Let's split this as two parts.
1st is making new binary, 2nd is downloading new binary.